beetsplug/export.py

```diff
@@ -108,44 +135,118 @@ def run(self, lib, opts, args):
             except (mediafile.UnreadableFileError, IOError) as ex:
                 self._log.error(u'cannot read file: {0}', ex)
                 continue
-
             data = key_filter(data)
             items += [data]
 
-        export_format.export(items, **format_options)
+        self.run_results = items
+        self.export_format.export(self.run_results, **format_options)
 
 
 class ExportFormat(object):
     """The output format type"""
+    def __init__(self, file_path, file_mode=u'w', encoding=u'utf-8'):
+        self.path = file_path
+        self.mode = file_mode
+        self.encoding = encoding
+        # Used for testing
+        self.results = None
 
     @classmethod
-    def factory(cls, type, **kwargs):
-        if type == "json":
-            if kwargs['file_path']:
-                return JsonFileFormat(**kwargs)
-            else:
-                return JsonPrintFormat()
-        raise NotImplementedError()
+    def factory(cls, file_type, **kwargs):
+        if file_type == "json":
+            return JsonFormat(**kwargs)
```

Comment:
I notice this now behaves differently from it did before—the `kwargs` are now forwarded unconditionally instead of only when the path is unspecified. Does this change the user-visible behavior?

Reply:
This should not effect the user in any way. It simply simplifies the logic in my opinion.